quinn-proto/src/connection/mod.rs

```diff
@@ -2747,6 +2737,40 @@ where
         sent
     }
 
+    /// Write pending ACKs into a buffer
+    ///
+    /// This method assumes ACKs are pending, and should only be called if
+    /// `PendingAcks::can_send()` or `!PendingAcks::ranges().is_empty()` returns `true`.
```

Comment:
Would it be clearer to omit mention of `PendingAcks::can_send()`, since it always implies `!PendingAcks::ranges().is_empty()`?

Reply:
Maybe - but it's debatable whether it should rather be the other way around and only `can_send()` should be checked in order to suppress always including all acks. But since that change isn't done here - and a reader of only that file might not see that `can_send` will check `!ranges().is_empy()`, I opted for putting both there. I would however be fine removing `can_send` for now if you think that's preferrable

Reply:
I think `can_send` isn't actually that strong--we use that to determine if we can send an ACK *only* packet, but we still encode ACKs regardless when sending ACK-eliciting packets. Conversely, `ranges()` is always sufficient to sanely call the method, even if we don't actually want to do that under all circumstances for other reasons.

Reply:
ok - updated the comment